CookingRecipies/Components/MyCookBook.js

```diff
@@ -0,0 +1,48 @@
+import React, {useState, useEffect} from "react";
+import 
+{View,TouchableOpacity, TextInput, Button, StyleSheet, Text} 
+from "react-native";
+
+
+const MyCookBook = (props) =>{
+    const[folderName, setFolderName] = useState([
+        {
+            course: '',
+            recipes:  []
+        }
+    ]);
+    const [savedRecipes,  setSavedRecipes] = useState([props]);
+
+    useEffect(()=>{
+        for(i in savedRecipes){
+            for(x in savedRecipses){
```

Comment:
@Obaida-Albaroudi you might need to reevaluate how this for loop is executing.  you're looping over the same data structure twice.